python/packages/litellm/agent_framework_litellm/_chat_client.py

```diff
@@ -0,0 +1,178 @@
+    with the encoding 'utf-8'. If the settings are not found in the .env file, the settings
+    are ignored; however, validation will fail alerting that the settings are missing.
+
+    Keyword Args:
+        model_id: The model to use, including both the provider and model (e.g., "azure_openai/gpt-4").
+        env_file_path: If provided, the .env settings are read from this file path location.
+        env_file_encoding: The encoding of the .env file, defaults to 'utf-8'.
+
+    Examples:
+        .. code-block:: python
+
+            from agent_framework_litellm import LiteLlmAISettings
+
+            # Using environment variables
+            # Set LITE_LLM_MODEL_ID=azure_openai/gpt-4
+            settings = LiteLlmAISettings()
+
+            # Or passing parameters directly
+            settings = LiteLlmAISettings(model_id="azure_openai/gpt-4")
+
+            # Or loading from a .env file
+            settings = LiteLlmAISettings(env_file_path="path/to/.env")
+    """
+
+    env_prefix: ClassVar[str] = "LITE_LLM_"
+
+    model_id: str | None = None
+
+
+logger = get_logger("agent_framework.litellm")
+
+
+@use_function_invocation
+@use_observability
+@use_chat_middleware
+class LiteLlmChatClient(OpenAIBaseChatClient):
+    """LiteLLM Chat client.
+
+    This client is used to interact with LiteLLM models via the Agent Framework. Note that LiteLLM is not fully OpenAI
+    API compatible, so some features may not be supported. However, LiteLLM follows the OpenAI API structure closely
+    enough to allow for basic interactions.
+    """
+
+    OTEL_PROVIDER_NAME: ClassVar[str] = "litellm"  # type: ignore[reportIncompatibleVariableOverride, misc]
+
+    def __init__(
+        self,
+        *,
+        api_key: str | None = None,
+        api_base: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = "utf-8",
+        model_id: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        self.api_key = api_key
+        self.api_base = api_base
+
+        try:
+            lite_llm_settings = LiteLlmCompletionAISettings(
+                model_id=model_id,
+                env_file_path=env_file_path,
+                env_file_encoding=env_file_encoding,
+            )
+        except ValidationError as ex:
+            raise ServiceInitializationError("Failed to create LiteLLM settings.", ex) from ex
+
+        if lite_llm_settings.model_id is None:
+            raise ServiceInitializationError(
+                "model_id parameter, or LITE_LLM_MODEL_ID env variable must be provided for LiteLlmChatClient"
+            )
+
+        self.model_id = str(lite_llm_settings.model_id)
+
+        super().__init__(api_base=api_base, api_key=api_key, model_id=self.model_id, client=None, **kwargs)  # type: ignore
+
+    def make_completion_streaming_request(self, **options_dict: Any) -> CustomStreamWrapper:
```

Comment:
these should probably be internal methods:

```suggestion
    def _make_completion_streaming_request(self, **options_dict: Any) -> CustomStreamWrapper:
```